device/cuda/include/traccc/cuda/utils/thread_id.hpp

```diff
@@ -8,34 +8,34 @@
 
 #pragma once
 
-#include <cstddef>
-
 #include "traccc/definitions/qualifiers.hpp"
 
 namespace traccc::cuda {
 struct thread_id1 {
     TRACCC_DEVICE thread_id1() {}
 
-    std::size_t inline TRACCC_DEVICE getLocalThreadId() const {
+    unsigned int inline TRACCC_DEVICE getLocalThreadId() const {
         return threadIdx.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getLocalThreadIdX() const {
+    unsigned int inline TRACCC_DEVICE getLocalThreadIdX() const {
         return threadIdx.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getGlobalThreadId() const {
+    unsigned int inline TRACCC_DEVICE getGlobalThreadId() const {
         return threadIdx.x + blockIdx.x * blockDim.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getGlobalThreadIdX() const {
+    unsigned int inline TRACCC_DEVICE getGlobalThreadIdX() const {
         return threadIdx.x + blockIdx.x * blockDim.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getBlockIdX() const { return blockIdx.x; }
+    unsigned int inline TRACCC_DEVICE getBlockIdX() const { return blockIdx.x; }
 
-    std::size_t inline TRACCC_DEVICE getBlockDimX() const { return blockDim.x; }
+    unsigned int inline TRACCC_DEVICE getBlockDimX() const {
+        return blockDim.x;
+    }
 
-    std::size_t inline TRACCC_DEVICE getGridDimX() const { return gridDim.x; }
+    unsigned int inline TRACCC_DEVICE getGridDimX() const { return gridDim.x; }
```

Comment:
? Why? :confused:

Reply:
These are all `unsigned int`-s natively. If we return them as `std::size_t`, I'll have to add a whole lot of `static_cast`-s in the kernels to make sure that we wouldn't use `unsigned long`-s where `unsigned int` is expected.

As it turns out, our CUDA build does not look for type conversions / narrowings. oneAPI does. So I got a **lot** of warnings from such things.

Reply:
Okay, but why not just add the `static_casts` then? `getGlobalThreadIdX()` can return a number larger than an unsigned integer can store.

Reply:
Also I am not sure why the OneAPI compiler even sees a type that is supposed to be used only in CUDA code. That's worrying. :worried:

Reply:
The kernels were/are receiving `std::size_t` values for the thread ID. And then they use those values in a lot of functions that expect `unsigned int` values. That's where I was getting the warnings from.

But as long as I'm modifying the signature of the `traccc::device::` functions, I thought I'd also have a look at what's going on in the CUDA code... 🤔